### PR TITLE
Md 722 geojson files route mismatch

### PR DESCRIPTION
**Pull request by xinaesthete**
The `mdvtools.spatial.conversion` script named geojson outputs with `.geojson` which then causes issues as the server would internally change this to `.ge.json` in the filename it was trying to serve.

## Summary by CodeRabbit

* **New Features**
  * Added dynamic per-project routing with project-specific routes and before-request hooks.
* **Chores**
  * GeoJSON output naming changed to use ".geo.json".
  * Added explanatory comments to a JSON endpoint handler (no behavior change).
* **Tests**
  * Added comprehensive routing tests covering pattern matching, converters, ordering, and edge cases.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: python/mdvtools/project_router.py [hunk 1]
```diff
@@ -59,48 +123,122 @@ def before_request(self, f: Callable) -> Callable:
...
-                        # If a hook returns a value, it becomes the response, short-circuiting the request
+                        # Hook returned a response - short-circuit and return it
+                        # (typically used for auth failures, redirects, etc.)
                         return rv
                 
-                # If all hooks passed (returned None), call the original view function
+                # All hooks passed - call the view function with captured groups
+                # TODO: Consider type conversion here (int, float, etc.) based on converter
+                # Currently just passes strings - handlers must convert themselves
                 return method(*match.groups())
```
[the remark this answers is not in the document]
Comment (xinaesthete) [reply python/mdvtools/project_router.py]: I think there have been other changes in `dbutils` related modules in another branch - we should maybe keep this flagged for review later, but changing the behaviour here seems risky.
